### ISSUE TITLE
Show the "published" side panel on every Update, not just Publish.

### ISSUE THREAD

**Issue by folletto** [WordPress/gutenberg#19749]
When I update a post I often want to do further actions, even just a simple "View post" to make sure everything reads well. So I appreciate having the button easy to reach right when the post is updated, together with the ability to easily copy the URL (there's often a reason when I update, and I tend to link back), and potentially other actions (from plugins), especially social media sharing if I did a relevant update. 

Structurally, I find odd that the "Publish" panel is shown only once, and no more for subsequent updates. It's an asymmetry that seems odd to me.

Also, minor but relevant frustrations:
* The "View post" link is shown just for a few seconds. As it's something I do all the time after posting/updating, if I miss it, I've now have to use a workaround to get to the post.
* The "View post" link is quite far away from the "Publish"/"Update" button.

### Bonus: Plugins

Having the post-publication side panel always showing up will make that space a more useful and relevant page for plugins, especially the ones that manage the "next step" of publication flows, either basic things like social media sharing, or more complex like editorial workflow management.

![Screenshot 2020-01-19 at 17 04 30](https://user-images.githubusercontent.com/4389/72685030-dd1bee00-3add-11ea-9076-6498aca9f485.png)

**Comment by youknowriad** [comment]
I do remember that this was the case initially and that the behavior changed based on user feedback. cc @jasmussen 

We have a few issues related to the publishing flow, it might be good to consider a rethinking in a single issue.

**Comment by jasmussen** [comment]
I suspect that user feedback was early frustration with a flow that was different from before (also leading us to add the checkbox at the bottom). I wonder if that has changed. 

This is an interesting idea, I wouldn't instinctually think this was a good idea; the pre publish flow is somewhat disruptive which is the intent when you're **publishing** something — in this day and age, making you think for two seconds before you actually publish, AND giving users an opportunity to check up last minute things, is the reason for it being disruptive. In that vein, adding this also on _update_ makes me pause.

However the reasoning is sound — "update" also publishes changes. 

I don't think "View Post" should be the reason to make this change, though — the fact that it comes up to me suggests a problem with the transient Snackbar. The transience is important for that component, it allows us to notify of small things that would otherwise be frustrating for a notice that needs dismissing, such as "Copied all content to clipboard" or other smaller things. Is there a use case for adding an optional close button to the snackbar, for items that should actually not be transient? I personally would love the successful publish notice to not be transient, and I can think of other situations like "Error saving, are you offline?" messages needing to be there until user interaction. If we added a close button we could theoretically — gasp! — retire the Notice component?

**Comment by folletto** [comment]
> I don't think "View Post" should be the reason to make this change, though

Yes, I agree. I thing the snackbar makes the problem more evident — but it's not the only bit that makes me think it's at least useful to reassess the side panel for this scenario :) 

> Is there a use case for adding an optional close button to the snackbar, for items that should actually not be transient?

I think there is, however, I'm not sure this is the scenario. I feel the side panel approach has many more benefits: it would solve thus, but it would also provide a lot more scalability. And everything at no added interaction cost, as the dismissal of a side panel is equivalent to the dismissal of any other component. It's one click, regardless, with the benefit that we already have a side panel design, we already show it, and it already has a close button that shows up very close to Publish :)

That said, if we decide that even after a review, the side pane is still not a good idea, I'd argue that the snackbar, specifically for "Update" has two issues:

1. Transiency — should be permanent, ideally with added a time, so I know WHEN the update was.
2. Placement  — right now it's at the opposite corner of the user action, which is imho a major interaction problem for this flow.

**Comment by jasmussen** [comment]
> Yes. I wanted to raise again as the original discussion was mostly based on hypotheticals, while now people had a long time to get the feel of the publication flow, and I feel it's now more easily visible how the initial fears weren't as worrying ;)

Seems worth a separate ticket to me. We force it on mobile for space reasons, it would add consistency and remove some code too.

> I think there is, however, I'm not sure this is the scenario. I feel the side panel approach has many more benefits: it would solve thus, but it would also provide a lot more scalability. And everything at no added interaction cost, as the dismissal of a side panel is equivalent to the dismissal of any other component. It's one click, regardless, with the benefit that we already have a side panel design, we already show it, and it already has a close button that shows up very close to Publish :)

We could do both! Though I suppose if we do the publish panel opening on update, the transiense of the "View Post" snackbar would be less important (but still something to think about separately).

> Placement — right now it's at the opposite corner of the user action, which is imho a major interaction problem for this flow.

Yep, we could probably revisit this too.

**Comment by folletto** [comment]
> Seems worth a separate ticket to me. We force it on mobile for space reasons, it would add consistency and remove some code too.

This is one more reason to review it, yes. :)
As for tickets: I'd keep this on the panel, seems more suited, and split out others if needed. :)

> Though I suppose if we do the publish panel opening on update, the transiense of the "View Post" snackbar would be less important (but still something to think about separately).

It probably can go away entirely at that point, as it would just replicate the content of the snackbar?

**Comment by jasmussen** [comment]
> It probably can go away entirely at that point, as it would just replicate the content of the snackbar?

I would think so, but worth double checking. I believe the snackbar is sent as an "alert", which makes screen readers read it aloud.

**Comment by folletto** [comment]
Which... it's something that the publish panel should be already doing, no? And if it does, the problem is automatically solved, if not, we probably should fix it for Publish?

**Comment by jasmussen** [comment]
Yep. Solid reasoning all around.

**Comment by Chrico** [comment]
Is there any update to that issue? I think enough time has passed to get some more input and feeling about the topic.

While the reasoning behind the "on publish (create) show some conditions and information before actually really publishing" makes totally sense, it could be the case that a PostType requires certain things (terms, meta, thumbnail, excerpt, ...) to be "full-filled" before it can go online. This is currently possible. But after the Post was published (created) you can just revert everything and bascially update the Post into a state which wouldn't be allowed as "published".

Doesn't it make sense to even provide a new `PreUpdatePanel,` which allows to do checks _before_ the Post can be updated?

**Comment by tyrann0us** [comment]
Hi @youknowriad, @aduth, @ellatrix, and @jasmussen, I'm very sorry to tag you directly, but I don't know of any other way to bring attention to this issue.

Given the length of time that has passed since the last discussion here (2 years), and given that @Chrico's comment has a considerable amount of reactions, I wonder if you could take another look at the issue? Thanks a lot!

**Comment by Marc-pi** [comment]
@ndiego / @annezazu / @jasmussen >> must be fixed in 6.2, it is a very annoying UX issue, brings friction and degrades the onboarding, writers loose time at each publishing/editing, it is systematic, not a purely cosmetic issue

from https://github.com/WordPress/gutenberg/issues/47446

+ there is indeed a different behavior between publish and update action
+ target blank missing on the View link url
+ confirmation msg is displayed at the opposite side (your eyes / mouse have to cross the whole screen to see the msg despite the source CTA is on the opposite side) > not a good UX practice at all

> Use Case : "i write my post, i want to see the final result / rendered content quickly, and be able to fine tune my content or fix some errors like spelling"

**Comment by Marc-pi** [comment]
i poke @audrasjb since it is an issue in between Gutenberg & Core (since he's looking for old issues to fix in his scrumb). This one is a very annoying one.

**Comment by Marc-pi** [comment]
partially adressed by @youknowriad / @jasmussen 
in https://github.com/WordPress/gutenberg/pull/47142

if you could bring consistency about the message box when saving within the site editor and edit create post/page.
would  ease new WP Gutenberg users onboarding !!
UX UX !
